hammr/commands/image/image.py

```diff
@@ -427,35 +399,39 @@ def do_download(self, args):
             # call UForge API
             printer.out("Searching image with id [" + doArgs.id + "] ...")
             images = self.get_all_images()
-            if images is None:
-                return
+            if len(images) == 0:
+                raise ValueError("No image available")
+
+            dlImage = None
+            for image in images:
+                if str(image.dbId) == str(doArgs.id):
+                    dlImage = image
+            if dlImage is not None and dlImage.status.complete and not dlImage.status.error and dlImage.compress:
+                download_url = self.api.getUrl() + "/" + dlImage.downloadUri
+                dlUtils = download_utils.Download(download_url, doArgs.file, not self.api.getDisableSslCertificateValidation())
+                try:
+                    dlUtils.start()
+                except Exception, e:
+                    return
+                printer.out("Image downloaded", printer.OK)
+            elif dlImage is None:
+                printer.out("Unable to find the image to download in your library", printer.ERROR)
```

Comment:
Maybe we can throw some exceptions here ? with the message inside ?

Reply:
I agree with you, but I did not touch this method at all, I think a refactor of this one could go in another pull request